### PR TITLE
roachtest: enable jemalloc profiling in all roachtests 

### PR DESCRIPTION
**Pull request by knz**

### CODE REVIEW

**Review by petermattis — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54260#-:-MGxi2BR4b0tWwC83x8J:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif, @knz, and @petermattis)

___
*[pkg/cmd/roachprod/install/cassandra.go, line 52 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54260#-MGxhpMx1o00gDqIhxQ8:-MGxhpMy03iiKt_e-7IA:b-gujv5x) ([raw file](https://github.com/cockroachdb/cockroach/blob/114eeb79a1b06a4ae5879aea79ae9a45ceb3095a/pkg/cmd/roachprod/install/cassandra.go#L52)):*
> ```Go
> 			}()
> 
> 			cmd := `env ` + c.Env + ` ROACHPROD=true cassandra` +
> ```

I wonder if we should nuke this cassandra support. I think it has been a long time since we've used it.

**Review by irfansharif — APPROVED** [review]

File: pkg/cmd/roachtest/cluster.go [hunk 1]
```diff
@@ -2121,6 +2121,10 @@ func (c *cluster) StartE(ctx context.Context, opts ...option) error {
 	args := []string{
 		roachprod,
 		"start",
+		// Enable Jemalloc profiles for all roachtests. This has only
+		// minimal performance imapct and significantly improves the
+		// troubleshootability of memory-related issues.
+		"-e", "MALLOC_CONF=prof:true",
```
Comment (irfansharif) [inline pkg/cmd/roachtest/cluster.go:2127]: [nit] Slight aesthetic preference for using a helper here and using `args = append(args, helperFn)` below.

Comment (knz) [reply pkg/cmd/roachtest/cluster.go:2127]: can you show me what you mean?

Comment (irfansharif) [reply pkg/cmd/roachtest/cluster.go:2127]:
```diff
diff --git i/pkg/cmd/roachtest/cluster.go w/pkg/cmd/roachtest/cluster.go
index 7d3bb3a8e4..025bb24d61 100644
--- i/pkg/cmd/roachtest/cluster.go
+++ w/pkg/cmd/roachtest/cluster.go
@@ -2124,12 +2124,17 @@ func (c *cluster) StartE(ctx context.Context, opts ...option) error {
        }
        args = append(args, roachprodArgs(opts)...)
        args = append(args, c.makeNodes(opts...))
+       args = append(args, enableJemallocArgs()...)
        if !argExists(args, "--encrypt") && c.encryptDefault {
                args = append(args, "--encrypt")
        }
        return execCmd(ctx, c.l, args...)
 }

+func enableJemallocArgs() []string {
+       return []string{"-e", "MALLOC_CONF=prof:true"}
+}
+
 // Start is like StartE() except it takes a test and, on error, calls t.Fatal().
 func (c *cluster) Start(ctx context.Context, t *test, opts ...option) {
        FatalIfErr(t, c.StartE(ctx, opts...))
```

Something like the above.

Comment (knz) [reply pkg/cmd/roachtest/cluster.go]: I don't understand it, but I trust you more than the code that was there before :)

Done

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54260)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @irfansharif and @petermattis)

___
*[pkg/cmd/roachprod/install/cassandra.go, line 52 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54260#-MGxhpMx1o00gDqIhxQ8:-MGxygaq2eKH3jb9Z10r:ba10cxg) ([raw file](https://github.com/cockroachdb/cockroach/blob/114eeb79a1b06a4ae5879aea79ae9a45ceb3095a/pkg/cmd/roachprod/install/cassandra.go#L52)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I wonder if we should nuke this cassandra support. I think it has been a long time since we've used it.
</blockquote></details>

I don't disagree but that seems a topic for a different PR.

**Comment by knz** [comment]
bors r=irfansharif,petermattis

**Pull request merged**